ksql-metastore/src/main/java/io/confluent/ksql/metastore/KsqlStream.java

```diff
@@ -41,43 +47,55 @@ public KsqlStream(
         DataSourceType.KSTREAM,
         ksqlTopic
     );
+    this.keySerde = Objects.requireNonNull(keySerde, "keySerde");
+  }
+
+  public boolean isWindowed() {
```

Comment:
Only `Tables` can be windowed and this will always return false.

Reply:
Also in near future we plan to support `struct` keys and relying on this condition for detecting windowed table won't be correct. So it's better to reduce relying on the string key serde as the condition of not being `table`.

Reply:
Yeah, I agree with the 'don't rely on String Serde as an indication its not windowed', though checking for window specific serdes might be sufficient.

As for only tables being windowed, see comments above.

Reply:
So are you making the change for checking window specific serde?